INSTALL.md

```diff
@@ -42,10 +42,11 @@ You can configure a FlexFlow build by running the `config/config.linux` file in
 3. `FF_CUDA_ARCH` is used to set the architecture of targeted GPUs, for example, the value can be 60 if the GPU architecture is Pascal. To build for more than one architecture, pass a list of comma separated values (e.g. `FF_CUDA_ARCH=70,75`). To compile FlexFlow for all GPU architectures that are detected on the machine, pass `FF_CUDA_ARCH=autodetect` (this is the default value, so you can also leave `FF_CUDA_ARCH` unset. If you want to build for all GPU architectures compatible with FlexFlow, pass `FF_CUDA_ARCH=all`. **If your machine does not have any GPU, you have to set FF_CUDA_ARCH to at least one valid architecture code (or `all`)**, since the compiler won't be able to detect the architecture(s) automatically.
 4. `FF_USE_PYTHON` controls whether to build the FlexFlow Python interface.
 5. `FF_USE_NCCL` controls whether to build FlexFlow with NCCL support. By default, it is set to ON.
-6. `FF_USE_GASNET` is used to enable distributed run of FlexFlow. If you want to run FlexFlow on multiple nodes, follow instructions in [MULTI-NODE.md](MULTI-NODE.md) and set `FF_USE_GASNET=ON` and `FF_GASNET_CONDUIT` as a specific conduit (e.g. `ibv`, `mpi`, `udp`, `ucx`) in `config/config.linux` when configuring the FlexFlow build.
-7. `FF_BUILD_EXAMPLES` controls whether to build all C++ example programs.
-8. `FF_MAX_DIM` is used to set the maximum dimension of tensors, by default it is set to 4. 
-9. `FF_USE_{NCCL,LEGION,ALL}_PRECOMPILED_LIBRARY`, controls whether to build FlexFlow using a pre-compiled version of the Legion, NCCL (if `FF_USE_NCCL` is `ON`), or both libraries . By default, `FF_USE_NCCL_PRECOMPILED_LIBRARY` and `FF_USE_LEGION_PRECOMPILED_LIBRARY` are both set to `ON`, allowing you to build FlexFlow faster. If you want to build Legion and NCCL from source, set them to `OFF`.
+6. `FF_USE_GASNET` is used to enable distributed run of FlexFlow with GASNet. If you want to run FlexFlow on multiple nodes by building FlexFlow with GASNet, follow instructions in [MULTI-NODE.md](MULTI-NODE.md) and set `FF_USE_GASNET=ON` and `FF_GASNET_CONDUIT` as a specific conduit (e.g. `ibv`, `mpi`, `udp`, `ucx`) in `config/config.linux` when configuring the FlexFlow build. Set `FF_UCX_URL` when you want to customize the URL to download UCX.
+7. `FF_LEGION_NETWORKS` is used to enable distributed run of FlexFlow without GASNet. You can follow instructions in [MULTI-NODE.md](MULTI-NODE.md) and set `FF_LEGION_NETWORKS` as `ucx` in `config/config.linux` when configuring the FlexFlow build.
```

Comment:
Let's deprecate `FF_USE_GASNET` and only use `FF_LEGION_NETWORKS`. `FF_LEGION_NETWORKS` can be either `ucx`, `gasnet` or `none`.